local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -15,12 +15,16 @@
 
 namespace avoidance {
 
-LocalPlannerNode::LocalPlannerNode() {
+LocalPlannerNode::LocalPlannerNode(const bool tf_spin_thread)
+    : tf_spin_thread_{tf_spin_thread} {
```

Comment:
You could just initialize the `tf_listener_` here, and then you don't need the `tf_spin_thread_` member variable in the class, and the `tf_listener_` doesn't need to be a pointer.